### PR TITLE
Make the prompt to upgrade dismissible

### PR DESCRIPTION
**Pull request by shervElmi**
### Issue link:
https://github.com/Strategy11/business-directory-premium/issues/103

### Changes proposed in this PR:
With this PR, we will make the Download Premium notice dismissible.

### How to test the changes in this Pull Request:
- Active one BD add-on without activating the Premium version.
- You need to dismiss the `"You have modules installed, but Business Directory Premium is missing. Install this plugin for extra features and easy license management."` notice in the BD admin panel and another page in wp-admin, like the Plugins page management.
- In both cases, this message should not be displayed after dismissing you.

### CODE REVIEW

**Review by garretlaxton — APPROVED** [review]
This looks good! It now stays dismissed after you click.

**Review by stephywells — CHANGES_REQUESTED** [review]
Nice fix! I just have one small request for the admin-wide js.

Once that is updated, will you go ahead and start on this checklist? https://app.asana.com/0/1189101947561289/1203292211866604/f

It's super important to have the release version PR merged before creating a release in Github. The Github release pushes to WP.org immediately, and if the version number is wrong, people will get infinite loops of update requests.

File: assets/js/wp-admin.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			dismissNotice( $notice, $button );
+		}
+	);
+
+	/**
+	 * Dismisses a notice.
+	 *
+	 * @param {HTMLElement} $notice The notice element.
+	 */
+	var dismissNotice = function ( $notice ) {
```
[the remark this answers is not in the document]
Comment (shervElmi) [reply assets/js/wp-admin.js:28]: The whole code is wrapped with invoked function! that means It prevents any conflict! but It's always suitable to prefix JavaScript functions to move them anywhere without worries. Thanks for your suggestion!

**Comment by shervElmi** [comment]
> Nice fix! I just have one small request for the admin-wide js.
> 
> Once that is updated, will you go ahead and start on this checklist? https://app.asana.com/0/1189101947561289/1203292211866604/f
> 
> It's super important to have the release version PR merged before creating a release in Github. The Github release pushes to WP.org immediately, and if the version number is wrong, people will get infinite loops of update requests.

Thanks!
I'll check it with you to make sure I'm doing it right.

**Review by stephywells — APPROVED** [review]

File: assets/js/wp-admin.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephywells) [reply assets/js/wp-admin.js:28]: Sounds good. You can go ahead and merge this PR then. Thanks!

**Comment by shervElmi** [comment]
@stephywells 

Would you please explain to me what I should do to fix this:
https://github.com/Strategy11/business-directory-plugin/actions/runs/3496885594/jobs/5855319519

Thanks!

**Comment by stephywells** [comment]
@shervElmi that doesn't require a fix. When only the readme is updated, that will push it to the wp.org repo immediately. But most of the time that won't be the case. The official releases are triggered when a new release is created here:
https://github.com/Strategy11/business-directory-plugin/releases

**Pull request merged**